doc/book/app_server/luajit_memprof.rst

```diff
@@ -0,0 +1,524 @@
+string via the ``concat()`` function, and LuaJIT just use the previous one.
+
+That is also the reason why the amount of allocations is not the round numbber
+as can be expected from the cycle operator ``for i = 1, 10000...``:
+Tarantool creates some
+strings for internal needs and built-in modules, so some strings already exist.
+
+But why are there so many allocations? It's almost twice as big as the expected
+amount. This is because the ``string.format()`` built-in function creates
+another string necessary for the ``%s`` identifier, so there are two allocations
+for each iteration: for ``tostring(i)`` and for ``string.format("%sa", string_i_value)``.
+You can see the difference in behaviour by adding the
+``local _ = tostring(i)`` line between lines 21 and 22.
+
+To profile only the ``concat()`` function, comment the line 22, namely,
+``local f = format(i)`` and run the profiler.
+
+The profiler's output is the following:
+
+..  code-block:: console
+
+    ALLOCATIONS
+    @format_concat.lua:3, line 4: 10000     284411  0
+
+    REALLOCATIONS
+
+    DEALLOCATIONS
+    INTERNAL: 10000 0       218905
+            Overrides:
+                    @format_concat.lua:3, line 4
+
+    @format_concat.lua:3, line 4: 1 0       32768
+
+
+**Q**: But what will change if the JIT compilation is enabled?
+
+**A**: Let's comment the first line of the code, namely, ``jit.off()`` and run
+the profiler . Now, there are only 56 allocations in the report, and all other
+allocations are JIT-related (see also the related
+`dev issue <https://github.com/tarantool/tarantool/issues/5679>`_):
+
+..  code-block:: console
+
+    ALLOCATIONS
+    @format_concat.lua:3, line 4: 56        1112    0
+    @format_concat.lua:0, line 0: 4 640     0
+    INTERNAL: 2     382     0
+
+    REALLOCATIONS
+
+    DEALLOCATIONS
+    INTERNAL: 58    0       1164
+            Overrides:
+                    @format_concat.lua:3, line 4
+                    INTERNAL
+
+This happens because a trace is compiled after 56 iterations, and the
+JIT-compiler removed the unused ``c`` variable  from the trace, and, therefore,
+the dead code of the ``concat()`` function is eliminated.
+
+Next, let's profile only the ``format()`` function with JIT enabled.
+For that, keep the lines 1 and 23 commented (``jit.off()`` and
+``local c = concat(i)`` respectively), uncomment the line 22
+(``local f = format(i)``), and run the profiler.
+
+The profiler's output is the following:
+
+..  code-block:: console
+
+    ALLOCATIONS
+    @format_concat.lua:8, line 9: 19998     624322  0
+    INTERNAL: 4     66824   0
+    @format_concat.lua:0, line 0: 4 640     0
+
+    REALLOCATIONS
+
+    DEALLOCATIONS
+    INTERNAL: 19999 0       559072
+            Overrides:
+                    @format_concat.lua:0, line 0
+                    @format_concat.lua:8, line 9
+
+    @format_concat.lua:8, line 9: 2 0       98304
+            Overrides:
+                    @format_concat.lua:8, line 9
```

Comment:
With the additional first line (see the comment above) the report looks like the following:
```
ALLOCATIONS
@format_concat.lua:9, line 10: 19998     624322  0
INTERNAL: 4     66824   0
@format_concat.lua:0, line 0: 4 640     0

REALLOCATIONS

DEALLOCATIONS
INTERNAL: 19999 0       559072
        Overrides:
                @format_concat.lua:0, line 0
                @format_concat.lua:9, line 10
@format_concat.lua:9, line 10: 2 0       98304
        Overrides:
                @format_concat.lua:9, line 10
```

Reply:
done